src/panel/category/CategoryPanel.jsx

```diff
@@ -41,17 +124,38 @@ export default class CategoryPanel extends React.Component {
     }
   }
 
+  close = () => {
+    SearchInput.setInputValue('');
+    window.app.navigateTo('/');
+  }
+
+  selectPoi = poi => {
+    fire('click_category_poi', poi, this.props.categoryName);
+  }
+
+  highlightPoiMarker = (poi, highlight) => {
+    fire('highlight_category_marker', poi, highlight);
+  }
+
   render() {
-    const { pois, dataSource, hasError, zoomIn, close, selectPoi, highlightPoiMarker } = this.props;
+    const { loading, pois, dataSource } = this.state;
+
+    if (loading) {
+      return null;
+    }
```

Comment:
As a consequence the panel will flicker when moving the map. Is it an expected behavior ?

Reply:
Indeed. I forgot a condition. Thanks!

Reply:
As we don't have any loading indicator for now and the only time when the panel is not displayed is during the first load, I changed this state boolean to `initialLoading`. Now it behaves like before.